app/assets/javascripts/initializers/initializeBaseUserData.js

```diff
@@ -108,6 +108,21 @@ function setCurrentUserToNavBar(user) {
       .getElementsByClassName('js-header-menu-admin-link')[0]
       .classList.remove('hidden');
   }
+
+  // If the user has policies in their data, let's iterate through them and start
+  if (user.policies) {
+    user.policies.forEach(function (policy) {
+      if (
+        policy.resource_type == 'Article' &&
+        policy.action == 'create' &&
+        !policy.allowed
+      ) {
+        document
+          .getElementsByClassName('js-policy-Article-create')[0]
+          .classList.add('hidden');
+      }
+    });
+  }
```

Comment:
I like that this could be very easily extended - everything which requires the ability to create articles could have this class name, and instead of accessing only the first matching item, we could iterate through all of them. A couple of thoughts though:

- @msarit touched on this, but hiding with CSS means that it's still in the elements panel, and it doesn't take much know-how to remove the 'hidden' class. This isn't a problem for features which are guarded in the backend too, but just something to consider. I see what we are proposing here as somewhat similar in spirit to [how we handle showing/hiding certain views in the mobile app webview](https://github.com/forem/forem/blob/main/app/assets/stylesheets/runtime.scss).
- Showing by default, then hiding via the JS is definitely the most 'progressive enhancement' way to do this, _but_ it means we are open to layout flashes. This particular link is hidden away in the dropdown menu, but if we applied this to the main "Create post" button, users with slower speeds would likely see a flash of the button before it disappeared, which isn't ideal. Maybe we want to consider whether in this scenario we do the opposite - hide by default, and remove the `hidden` class when appropriate in the JS
- In some places, I think we would see cumulative layout shift with this approach (in places where the element isn't hidden in a dropdown, for example), as an element either appears or is hidden. CLS can result in poorer SEO, and we try to avoid it wherever possible (this is why, for example, we show "Follow" buttons before they are actually initialized or updated to have the correct text inside them). 

Another thought I had was "can we make this even simpler?". What if the `base_data` policies property was an object with a simple 'can this user view this class' boolean, e.g.

```json
{
  "policies": {
    "js-policy-article-create": false
  }
}
```

Then the JS code only needs to iterate through each class, showing or hiding as appropriate. There is likely some considerations I'm missing here, but it just felt like it could be very extensible instead of having to check e.g. ` policy.resource_type == 'Article'` and so on, which could become unwieldy as other policies are added

Reply:
@aitchiss thank you for all this context, the CLS consideration is something that will require a trial and error effort.

And regarding "making it even simpler", last night I was all like: _Wait? I have can make a Ruby method that will render the proper CSS class for the policy.  And since I'm rendering that JSON in Ruby, why don't I repurpose that method? Then, the Javascript will iterate over policies and then iterate over each class and toggle accordingly._

I don't want to go too far down the CLS until we see more of what we're working against.  But I'd imagine we could consider transitioning from disabled to enabled (or vice versa).

Reply:
>last night I was all like: Wait? I have can make a Ruby method that will render the proper CSS class for the policy. And since I'm rendering that JSON in Ruby, why don't I repurpose that method? Then, the Javascript will iterate over policies and then iterate over each class and toggle accordingly

🧠  ✨  sounds _ideal_

> I don't want to go too far down the CLS until we see more of what we're working against. 

Yeah depending where things are, we can consider workarounds. In some cases it might be appropriate to "reserve" space so the layout stays consistent, and like you say maybe disabled/enabled is possible in some contexts too.